dcos-log/mesos/files/reader/read.go

```diff
@@ -103,58 +108,68 @@ func NewLineReader(client *http.Client, masterURL url.URL, agentID, frameworkID,
 			length = rm.offset
 		}
 
-		// TODO: split this function into smaller ones to improve the readability.
-		for {
-			ctx, cancel := context.WithTimeout(context.Background(), time.Second*3)
-			lines, delta, err := rm.read(ctx, offset, length, reverse)
-			if err != nil {
-				cancel()
-				return nil, err
-			}
+		err := calcOffset(offset, length, rm)
+		if err != nil && err != io.EOF {
```

Comment:
why does `io.EOF` get a pass here? because we keep monitoring at some point and start returning data eventually?

Reply:
when we read backwards, we will call rm.read() eventually which returns io.EOF, this is actually a bug fix for something that we found in CLI integration tests